### PR TITLE
feat: Wrapper for messages

### PR DESCRIPTION
**Pull request by caalador**
part of #1371

### CODE REVIEW

File: vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/messages/MessageListWrap.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+ * 4.0 (CVDLv4).
+ *
+ *
+ * For the full License, see <https://vaadin.com/license/cvdl-4.0>.
+ */
+package com.vaadin.flow.component.messages;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/messages/MessageListWrap.java:14]: Unused imports
```suggestion
```

File: vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/messages/MessageListWrap.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+     *
+     * @return all available messages
+     */
+    public List<MessageListItem> getMessages() {
+        ensureComponentIsUsable();
+        return getComponent().getItems();
+    }
+
+    /**
+     * Get messages between given Instances (excluding).
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/messages/MessageListWrap.java:91]: maybe a typo
```suggestion
     * Get messages between given instants (excluding).
```

File: vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/messages/MessageListWrap.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+     *
+     * @param start
+     *            start time
+     * @param end
+     *            end time
+     * @return massages falling between start an end time
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/messages/MessageListWrap.java:97]:
```suggestion
     * @return messages falling between start an end time
```

File: vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/messages/MessageListWrap.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    public List<MessageListItem> getMessagesAfter(Instant start) {
+        ensureComponentIsUsable();
+        return getComponent().getItems().stream()
+                .filter(message -> message.getTime() != null)
+                .filter(message -> message.getTime().isAfter(start))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Get messages before given Instance (excluding).
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/messages/MessageListWrap.java:124]: maybe a typo
```suggestion
     * Get messages before given instant (excluding).
```

File: vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/messages/MessagesView.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * This program is available under Commercial Vaadin Developer License
+ * 4.0 (CVDLv4).
+ *
+ *
+ * For the full License, see <https://vaadin.com/license/cvdl-4.0>.
+ */
+package com.vaadin.flow.component.messages;
+
+import java.time.Instant;
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/messages/MessagesView.java:12]: Unused import
```suggestion
```

File: vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/messages/MessageListWrapTest.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.Extensions;
```
Comment (mcollovati) [inline vaadin-testbench-unit/src/test/java/com/vaadin/flow/component/messages/MessageListWrapTest.java:20]: Unused import
```suggestion
```

File: vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/messages/MessageInputWrap.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    /**
+     * Send an input message through the MessageInput component.
+     *
+     * @param message
+     *            message from component
+     */
+    public void send(String message) {
+        ensureComponentIsUsable();
+        ComponentUtil.fireEvent(getComponent(),
```
[the remark this answers is not in the document]
Comment (caalador) [reply vaadin-testbench-unit/src/main/java/com/vaadin/flow/component/messages/MessageInputWrap.java:45]: Empty or null messages do not fire event now

**Review by mcollovati — APPROVED** [review]

**Pull request merged**